### PR TITLE
top: make output tabular

### PR DESCRIPTION
**Pull request by vrothberg**
Make the output of top tabular to be compatible with Docker.

Fixes: #458
Signed-off-by: Valentin Rothberg <vrothberg@suse.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@baude @mheon I believe there was a reason why we did not do this?

**Comment by vrothberg** [comment]
@rhatdan Based on #458, the main concern was that some columns can contain whitespaces, so I had a look at what Moby does (as mentioned here https://github.com/projectatomic/libpod/issues/458#issuecomment-396838642) and it turns out that basic assumption is that `CMD` is always the last column and all exceeding ones get merged into that.

**Comment by mheon** [comment]
Can we guarantee that `CMD` is the last entry? Can we validate that it is somewhere? Because if we can, I'd be happy to get this in, the `top` API is really gross right now.

**Comment by vrothberg** [comment]
> Can we guarantee that CMD is the last entry? Can we validate that it is somewhere? Because if we can, I'd be happy to get this in, the top API is really gross right now.

Yes, we can guarantee it as we specify the ps output explicitly just as Moby is doing as well.

**Comment by mheon** [comment]
I am fully in favor of forcing some structure on the arguments we pass into `ps`.

**Comment by vrothberg** [comment]
@mheon, we do it here https://github.com/vrothberg/libpod/blob/e58d97251ea28ae7d6e2d118e6e508d6999c8862/cmd/podman/top.go#L40

**Comment by vrothberg** [comment]
@mheon I would personally prefer to just hard-code the arguments into `GetContainerPidInformation`. If a user wants a different output, then there is still the option to `exec`. The main issue is that `ps` doesn't have a very friendly output format as grep for instance, where we can split columns with \0.

**Comment by rhatdan** [comment]
I agree lets remove the flexibility here and force the users to exec into the container to execute the ps command.

**Comment by baude** [comment]
great work @vrothberg !  thanks for doing this ... LGTM

**Comment by rhatdan** [comment]
I will leave this up to @mheon and @vrothberg to figure out his command about CMD.  ANd then merge.

**Comment by vrothberg** [comment]
I just updated the PR. It took a bit longer but I wanted to have a closer look at Moby. @mheon and I have agreed on ignoring the argument in `GetContainerPidInformation(args []string)` and just hard-code the ps argument until we have figured out a more generic and reliable way of extracting the data.

The basic problem is that ps doesn't support a way of splitting columns in a way to reliably parse the output and Moby/Docker seems to struggle as well. @mheon and I were dreaming about a library in go that does all the procFS parsing, but there is none yet (to the best of my knowledge). I may find some time next month during Hackweek to look at it.

Also I changed the ps args to `-ef` as Moby does. I noticed that the UID in docker-top reports root, which doesn't work with the `uid` format flag which always uses the numerical ID.

File: test/e2e/top_test.go [hunk 1]
```diff
@@ -59,26 +59,32 @@ var _ = Describe("Podman top", func() {
...
-		result.WaitWithDefaultTimeout()
-		Expect(result.ExitCode()).To(Equal(125))
-	})
+	//      XXX(ps-issue): for the time being, podman-top and the libpod API
+	//      GetContainerPidInformation(...) will ignore any arguments passed to ps,
+	//      so we have to disable the tests below.  Please refer to
+	//      https://github.com/projectatomic/libpod/pull/939 for more background
+	//      information.
+	//
+	//	It("podman top with options", func() {
```
Comment (mheon) [inline test/e2e/top_test.go]: Instead of commenting these out, can you add `Skip("reason")` as the first line of the test?

**Comment by mheon** [comment]
One nit otherwise LGTM

**Comment by vrothberg** [comment]
@mheon Done :+1: Thanks for the reviews!

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**